polaris-service/src/test/java/org/apache/polaris/service/catalog/TestUtil.java

```diff
@@ -140,7 +140,10 @@ public static RESTCatalog createSnowmanManagedCatalog(
             .header("Authorization", "Bearer " + adminToken.token())
             .header(REALM_PROPERTY_KEY, realm)
             .get()) {
-      assertThat(response).returns(Response.Status.OK.getStatusCode(), Response::getStatus);
+      // Buffer the entity so we can read it multiple times
+      response.bufferEntity();
```

Comment:
Seems like we read the response multiple times for all the assertion lines, not just this one -- should this be inside the `assertStatusCodeWithFailMessage` helper method?

I guess maybe we get away with it if the entity is an empty body for CREATED response codes, but generally it'd still be better if the caller doesn't have to know about whether or not the helper method "needs" the response to have `bufferEntity()` based on assumed contents.

Reply:
That makes sense. Agree on the point that the caller of the helper method is not responsible of buffering the entity since the helper method should operate on the assumption that the response entity would be read multiple times. Updated the code to accommodate this.